sse2neon.h

```diff
@@ -1693,6 +1708,39 @@ FORCE_INLINE __m128 _mm_mul_ps(__m128 a, __m128 b)
         vmulq_f32(vreinterpretq_f32_m128(a), vreinterpretq_f32_m128(b)));
 }
 
+// Multiply the low unsigned 32-bit integers from each packed 64-bit element in
+// a and b, and store the unsigned 64-bit results in dst.
+//
+//   r0 :=  (uint32_t*)a0 * (uint32_t*)b0
+//   r1 :=  (uint32_t*)a3 * (uint32_t*)b3
+#if 1 /* C version */
+FORCE_INLINE __m128i _mm_mul_epu32(__m128i a, __m128i b)
+{
+    __m128i d;
+    vreinterpretq_nth_u64_m128i(d, 0) =
+        (uint64_t)(vreinterpretq_nth_u32_m128i(a, 0)) *
+        (uint64_t)(vreinterpretq_nth_u32_m128i(b, 0));
+    vreinterpretq_nth_u64_m128i(d, 1) =
+        (uint64_t)(vreinterpretq_nth_u32_m128i(a, 2)) *
+        (uint64_t)(vreinterpretq_nth_u32_m128i(b, 2));
```

Comment:
How about using [vmull_*](https://developer.arm.com/architectures/instruction-sets/simd-isas/neon/intrinsics?search=vmull_) instruction?
I guess `vmull_u32` would work.
It uses [**UMULL**](https://developer.arm.com/docs/ddi0596/a/a64-simd-and-floating-point-instructions-alphabetic-order/umull-umull2-vector-unsigned-multiply-long-vector) instruction to extract each source vector from the lower half of each source register.